resolwe_bio/processes/workflows/bbduk_star_featurecounts_qc.py

```diff
@@ -0,0 +1,802 @@
+"""General RNA-seq pipeline."""
+
+from resolwe.process import (
+    BooleanField,
+    Data,
+    DataField,
+    FloatField,
+    GroupField,
+    IntegerField,
+    ListField,
+    Process,
+    StringField,
+)
+from resolwe.process.models import Process as BioProcess
+
+
+class WorkflowBBDukStarFcQC(Process):
+    """RNA-seq pipeline comprised of preprocessing, alignment and quantification.
+
+    First, reads are preprocessed by __BBDuk__ which removes adapters, trims
+    reads for quality from the 3'-end, and discards reads that are too short
+    after trimming. Compared to similar tools, BBDuk is regarded for its
+    computational efficiency.  Next, preprocessed reads are aligned by __STAR__
+    aligner. At the time of implementation, STAR is considered a
+    state-of-the-art tool that consistently produces accurate results from
+    diverse sets of reads, and performs well even with default settings. For
+    more information see [this comparison of RNA-seq
+    aligners](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5792058/). Finally,
+    aligned reads are summarized to genes by __featureCounts__. Gaining wide
+    adoption among the bioinformatics community, featureCounts yields
+    expressions in a computationally efficient manner. All three tools in
+    this workflow support parallelization to accelerate the analysis.
+
+    rRNA contamination rate in the sample is determined using the STAR aligner.
+    Quality-trimmed reads are down-sampled (using Seqtk tool) and aligned to the
+    rRNA reference sequences. The alignment rate indicates the percentage of the
+    reads in the sample that are derived from the rRNA sequences.
+    """
+
+    slug = "workflow-bbduk-star-featurecounts-qc"
+    name = "BBDuk - STAR - featureCounts - QC"
+    requirements = {
+        "expression-engine": "jinja",
+    }
+    data_name = "{{ preprocessing.reads|sample_name|default('?') }}"
+    version = "5.0.0"
+    process_type = "data:workflow:rnaseq:featurecounts:qc"
+    category = "Pipeline"
+
+    class Input:
+        """Input fields."""
+
+        class Preprocessing:
+            """Preprocessing with BBDuk."""
+
+            reads = DataField(
```

Comment:
I suggest to exclude the required inputs (reads, genome, etc) from the corresponding parent groups and leave them on the top level. List the groups of non-required inputs after the required inputs.  Boolean field `show_advanced` can then be omitted.